src/sync/sync_manager.hpp

```diff
@@ -159,18 +159,28 @@ friend class SyncSession;
 
     // Get a sync user for a given identity, or create one if none exists yet, and set its token.
     // If a logged-out user exists, it will marked as logged back in.
-    std::shared_ptr<SyncUser> get_user(const SyncUserIdentifier& identifier,
+    std::shared_ptr<SyncUser> get_user(const std::string& id,
```

Comment:
Why are we adding the provider_type to all of these methods? From my understanding, each user has a unique identifier which should be enough? The provider should only be used when logging in, right?

Reply:
If user_id+provider_type actually is needed to identify users, then shouldn't set_current_user() also need both? These functions seem to handle the case where two different providers have users with the same user_id (and those are actually different users that need to be distinguished), but the current user persistence is only storing the user_id.

Reply:
This was my mistake that I have since clarified with Stitch and update the PR. We need `provider_type` because only anon-user's are removed on logout (and it can be useful for users), but the `user_id` is actually unique across providers, so we do not need to identity users by user_id+provider_type– just user_id.